### PR TITLE
SDK (AsyncSubtensor) Part 1

### PR DESCRIPTION
**Pull request by basfroman**
Included changes:
- bittensor.core.async_subtensor.AsyncSubtensor class
-  updates settings, utils, etc
- added bittensor.utils.delegates_details.DelegatesDetails
- updated bittensor/core/chain_data
- added bittensor/core/extrinsics/async_registration
- added bittensor/core/extrinsics/async_root
- added bittensor/core/extrinsics/async_transfer

All extrinsics added to AsyncSubtensor as methods

### CODE REVIEW

File: bittensor/core/extrinsics/async_transfer.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            console.print(f":cross_mark: [red]Failed[/red]: {err_msg}")
+
+    if success:
+        with console.status(":satellite: Checking Balance...", spinner="aesthetic"):
+            new_balance = await subtensor.get_balance(
+                wallet.coldkeypub.ss58_address, reuse_block=False
+            )
+            console.print(
+                f"Balance:\n"
+                f"  [blue]{account_balance}[/blue] :arrow_right: [green]{new_balance[wallet.coldkey.ss58_address]}[/green]"
```
Comment (thewhaleking) [inline bittensor/core/extrinsics/async_transfer.py]:
```suggestion
                f"  [blue]{account_balance}[/blue] :arrow_right: [green]{new_balance[wallet.coldkeypub.ss58_address]}[/green]"
```

File: bittensor/core/extrinsics/async_registration.py [hunk 2]
```diff
@@ -0,0 +1,1618 @@
...
+    if not await subtensor.subnet_exists(netuid):
+        err_console.print(
+            f":cross_mark: [red]Failed[/red]: error: [bold white]subnet:{netuid}[/bold white] does not exist."
+        )
+        return False
+
+    with console.status(
+        f":satellite: Checking Account on [bold]subnet:{netuid}[/bold]...",
+        spinner="aesthetic",
+    ) as status:
```
Comment (thewhaleking) [inline bittensor/core/extrinsics/async_registration.py]: Do we want this stuff in the SDK?

File: bittensor/core/extrinsics/async_registration.py [hunk 3]
```diff
@@ -0,0 +1,1618 @@
...
+        # pow failed
+        if not pow_result:
+            # might be registered already on this subnet
+            is_registered = await is_hotkey_registered(
+                subtensor, netuid=netuid, hotkey_ss58=wallet.hotkey.ss58_address
+            )
+            if is_registered:
+                err_console.print(
+                    f":white_heavy_check_mark: [green]Already registered on netuid:{netuid}[/green]"
+                )
```
Comment (thewhaleking) [inline bittensor/core/extrinsics/async_registration.py]: Shouldn't these be do with btlogging?

File: bittensor/core/settings.py [hunk 4]
```diff
@@ -53,30 +53,65 @@ def turn_console_on():
...
+
+
+def print_error(message: str, status=None):
+    """Print error messages while temporarily pausing the status spinner."""
+    if status:
+        status.stop()
+        print_console(message, "red", "Error", bt_err_console)
+        status.start()
+    else:
+        print_console(message, "red", "Error", bt_err_console)
```
Comment (thewhaleking) [inline bittensor/core/settings.py]: Does this make sense in an SDK?

**Review by thewhaleking — CHANGES_REQUESTED** [review]
Few initial change requests and questions. Most of the questions are more high-level so I want to make sure we're all on the same page first.

File: bittensor/core/settings.py (hunk 4 above)
Comment (roman-opentensor) [reply bittensor/core/settings.py:83]: At this step/part, I want to make everything work and not break the async logic that I took from the btcli and changed a bit. 
Now any `console` based functions works and performs the same functions as btlogging. I can do this refactoring either in the next step or during the creation of the common package. I would answer both remaining comments this way.

Comment (basfroman) [reply bittensor/core/settings.py]: After some thought, I applied all the comments.

NOTE: Do not update the branch until the `testnet` is updated from `devnet`

Comment (basfroman) [reply bittensor/core/settings.py]: Merge after https://github.com/opentensor/bittensor/pull/2377 is merged

**Review by thewhaleking — APPROVED** [review]
Approved pending merge conflict

**Pull request merged**